src/user.controller.ts

```diff
@@ -118,29 +120,127 @@ export class UserController {
         // Create a Paginated query to return all the users
         const allUsers: [User] = await this.repository
             .createQueryBuilder('user')
-            .select(['user.id', 'user.firstName', 'user.lastName', 'user.email'])
+            .select(['user.id', 'user.firstName', 'user.lastName', 'user.email', 'user.enabled'])
             .skip(request.offset)
             .take(request.limit)
             .where('user.email like :email and user.firstName like :firstName and user.lastName like :lastName', filter)
             .getMany();
 
-            // Count how many records match
-            const userSize: any = await this.repository
-              .createQueryBuilder('user')
-              .select('COUNT(user.id)', 'count')
-              .where('user.email like :email and user.firstName like :firstName and user.lastName like :lastName', filter)
-              .getRawOne();
-        
+        // Count how many records match
+        const userSize: any = await this.repository
+            .createQueryBuilder('user')
+            .select('COUNT(user.id)', 'count')
+            .where('user.email like :email and user.firstName like :firstName and user.lastName like :lastName', filter)
+            .getRawOne();
+
+        return {
+            success: true,
+            message: 'All users',
+            data: {
+                users: allUsers,
+                count: userSize.count,
+            },
+        };
+
+    }
+
+    @Post('/update')
+    @ContentType("application/json")
+    @OpenAPI({ summary: 'Update an user' })
+    @ResponseSchema(Response, {
+        contentType: 'application/json',
+        statusCode: '200',
+    })
+    public async updateUser(@Body({ type: UpdateUserRequest }) request: UpdateUserRequest): Promise<Response> {
```

Comment:
We should check if the user exists before doing an update, and if not return that information on a message

Reply:
Done. I'd make a condition in (user) after the query, if exists, then update the user, else just throw a message, also return the success var in false.